### PR TITLE
add command to remove legacy plugin

### PR DESCRIPTION
**Pull request by ktalebian**
* Adds a `--remove-legacy-plugin` to remove legacy plugin
* Adds warning to deploy if legacy plugin is remaining

### CODE REVIEW

File: src/clients/FlexConfigurationClient.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  public async fetch(): Promise<ConfigurationInstance> {
+    return this.client.fetch();
+  }
+
+  /**
+   * Fetches the Serverless ServiceSid
+   */
+  public async getServerlessSid(): Promise<string | null> {
+    const config = await this.fetch();
+    if (!config.serverlessServiceSids) {
```
Comment (ahcai) [inline src/clients/FlexConfigurationClient.ts:25]: you can just `return config?.serverlessServiceSids[0];`

File: src/prints/upgradePlugin.ts [hunk 2]
```diff
@@ -106,6 +106,66 @@ const warnNotRemoved = (logger: Logger) => (note: string) => {
...
+const warningPluginNotInAPI = (logger: Logger) => (pluginName: string) => {
+  const name = coloredStrings.name(pluginName);
+
+  logger.info(`Plugin ${name} has not been migrated to Plugins API.`);
+  logger.newline();
+  logger.info(`Run {{$ twilio flex:plugins:upgrade-plugin \\-\\-beta \\-\\-install}} to upgrade your plugin code.`);
+  logger.info(
+    `Run {{$ twilio flex:plugins:deploy \\-\\-changelog "migrating to Flex Plugins API" \\-\\-major}} to register with Plugins API.`,
+  );
+  logger.info(`Run {{$ twilio flex:plugins:upgrade-plugin --remove-legacy-plugin}} again after to finish migration.`);
```
Comment (ahcai) [inline src/prints/upgradePlugin.ts:166]: does this need the `\\` like the ones above? (also why is that necessary)

**Review by ahcai — APPROVED** [review]

**Review by krishnakalluri — APPROVED** [review]

File: src/prints/upgradePlugin.ts (hunk 2 above)
Comment (ktalebian) [reply src/prints/upgradePlugin.ts:166]: Only need the escape if we have even flags (because `--the text in red--` syntax we have. So if you were to do `command --flag1 flagValue --flag2 flagValue2` then the logger would think `--flag1 flagValue --` is the text to color in red)

File: src/clients/FlexConfigurationClient.ts (hunk 1 above)
Comment (ktalebian) [reply src/clients/FlexConfigurationClient.ts:25]: TS didn't like 🤷
